### DISCUSSION TITLE
Can I load multiple plugins and then configure them afterwards?

### DISCUSSION THREAD

**Discussion by Hubro** [wbthomason/packer.nvim#500] [Q&A] [answered]
I want to specify a specific set of plugins, then run a Lua function when they have all been loaded. I hoped this would work:

```lua
  use {
    "rakr/vim-one",                -- One theme (dark and light)
    "tomasr/molokai",              -- Molokai
    "arcticicestudio/nord-vim",    -- Nord
    "cocopon/iceberg.vim",         -- Iceberg
    "morhetz/gruvbox",             -- Gruvbox
    "ayu-theme/ayu-vim",           -- Ayu
    "folke/tokyonight.nvim",       -- TokyoNight

    config = function()
      vim.exec "colorscheme gruvbox"
    end
  }
```

This installs all the color schemes, but the config function never runs.

I don't want to just place `colorscheme gruvbox` after `packer.startup(...)`, because that will crash if the color schemes haven't been installed yet. I could do a `try ... catch`, but that feels like a hack/workaround.

**Comment by wbthomason** [comment, marked as answer]
The typical way you'd do this is to change your spec to
```lua
use {
    "rakr/vim-one",                -- One theme (dark and light)
    "tomasr/molokai",              -- Molokai
    "arcticicestudio/nord-vim",    -- Nord
    "cocopon/iceberg.vim",         -- Iceberg
    {"morhetz/gruvbox",             -- Gruvbox
    config = function()
      vim.exec "colorscheme gruvbox"
    end},
    "ayu-theme/ayu-vim",           -- Ayu
    "folke/tokyonight.nvim",       -- TokyoNight
  }
```
because `config` keys are associated with single plugins.

**Reply by Hubro** [reply]
Kind of a pain to change the default color scheme though... If it's not possible to write one config function for a group of plugins, perhaps `try / catch` is the best solution.

**Reply by wbthomason** [reply]
Yeah, that's fair. Right now, `packer` just doesn't have a semantically meaningful notion of a group of plugins associated with a single `config` (the closest we come is a `requires` or `after` grouping). Personally, I just set my colorscheme in my `init.lua`.

#246 might be a relevant PR? If we can define a clear, sensible meaning for a plugin group, I'm not against adding the functionality.

Also, note that #508 will prevent `config` from running if its plugin hasn't been installed yet.

**Reply by wbthomason** [reply]
Sorry, I meant to reference #256, which is more germane than #246.
